oteapi/strategies/parse/postgres.py

```diff
@@ -35,78 +33,78 @@ class PostgresResourceConfig(ResourceConfig):
         description="Configuration options for the local data cache.",
     )
 
+    @model_validator(mode="before")
     @classmethod
-    def _urlconstruct(
```

Comment:
Why was this section removed? @quaat  was a key architect of this section so you should check a bit with him before removing

Reply:
Because the construction is now implemented as a part of the type. This can be done inherently with pydantic and utilizing the `typing.Annotated` class.
In any case, I'll add @quaat as a reviewer of the changes in this file as well 👍

Reply:
So just an additional comment here. The construction of the URL can be done with the `AnyUrl`'s `build()` class method. This was also the case in pydantic v1.
For pydantic v2 one can go further and add any constraints to a url by using `typing.Annotated` and pydantic v2's new `Url` and `UrlConstraints` classes. Furthermore, a `PostgresUrl` class already existed in pydantic's arsenal. But I suppose there was a desire to support `http`/`https` url schemes in this strategy as well, hence it was not used? In any case, this is easily done now - however, the specific checks and double checks done for each part of the URL that is being done here, is not possible with this - only data type specification and validation is possible, hence this approach is not used for `accessUrl` in this case (but it is used now for other strategies to replace subclassing `AnyUrl`).